akka-stream/src/main/scala/akka/stream/impl/fusing/FlatMapPrefix.scala

```diff
@@ -26,28 +26,28 @@ import scala.util.control.NonFatal
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[M]) = {
     val matPromise = Promise[M]
     val logic = new GraphStageLogic(shape) with InHandler with OutHandler {
-      var accumulated = List.empty[In]
+      var accumulated = (List.empty[In], 0)
```

Comment:
I'd still appreciate using a buffer here as suggested before. That will use less allocations and will make the code better readable.